pkgs/development/python-modules/pysqlite3/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, python
+, sqlite
+}:
+
+buildPythonPackage rec {
+  pname = "pysqlite3";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "coleifer";
+    repo = "pysqlite3";
+    rev = "${version}";
+    hash = "sha256-Ik//afKc7v1iBRGwOZrQbbMsHmdH5FptS9EAldhKRmk=";
+  };
+
+  buildInputs = [
+    sqlite
+  ];
+
+  postInstall = ''
+    rm -rf $out/${python.sitePackages}/tests
+  '';
+
```

Comment:
You are missing a pythonImportsCheck and running the tests.

Reply:
Should I insert only `pythonImportsCheck = [ "pysqlite3" ];`?

Reply:
No, I think you need to add `pythonImportCheckHook` and `pythonImportsCheckHook` into the `nativeBuildInputs`.

Reply:
Fixed. I added only `pythonImportsCheckHook ` because `pythonImportCheckHook ` does not exist.

Reply:
Oh, sorry, copy-paste error.

Reply:
I also meant to suggest `pytestCheckHook`.

Reply:
Fixed @KAction

Reply:
Does it build for you? Because for me it collects 0 tests and fails.

After more careful looking, it seems that upstream uses `unittest`, so you need `unittestCheckHook`.
It complains about `No module named 'pysqlite3._sqlite3'`, so you will need to figure out how to fix that.

Reply:
@KAction I implemented `unittestCheckHook` and I tried to use this:
```nix
  postPatch = ''
    substituteInPlace pysqlite3/dbapi2.py \
      --replace pysqlite3._sqlite3 _sqlite3
  '';
```
As output I get:
```
Ran 0 tests in 0.000s

OK
Finished executing unittestCheckPhase
/nix/store/0cjafmr1wmnqk6qw43s9vfbrqpdp8m0a-python3.11-pysqlite3-0.5.2
```
but I don't get error. Despite I get 0 tests, is it good?

Reply:
I don't think so. https://github.com/coleifer/pysqlite3/tree/master/test looks like it contains bunch of tests.

Reply:
> I don't think so. https://github.com/coleifer/pysqlite3/tree/master/test looks like it contains bunch of tests.

By reading build instructions of pysqlite3 in their README, it needs to be copied `sqlite3.c` and `sqlite3.h`. I did that and indeed, during the building, I get the message:
```sh
Executing setuptoolsBuildPhase
running build_ext
Builds a C extension linking against libsqlite3 library
building 'pysqlite3._sqlite3' extension
...
```
with no error, and then, when it runs the test, we get our infamous error:
```sh
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: pysqlite3
unittestCheckPhase
Executing unittestCheckPhase
E
======================================================================
ERROR: pysqlite3 (unittest.loader._FailedTest.pysqlite3)
----------------------------------------------------------------------
ImportError: Failed to import test module: pysqlite3
Traceback (most recent call last):
  File "/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/unittest/loader.py", line 452, in _find_test_path
    package = self._get_module_from_name(name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/unittest/loader.py", line 362, in _get_module_from_name
    __import__(name)
  File "/build/source/pysqlite3/__init__.py", line 23, in <module>
    from pysqlite3.dbapi2 import *
  File "/build/source/pysqlite3/dbapi2.py", line 28, in <module>
    from pysqlite3._sqlite3 import *
ModuleNotFoundError: No module named 'pysqlite3._sqlite3'


----------------------------------------------------------------------
Ran 1 test in 0.000s
```
So I start to think that maybe we need to use an environment variable to be set or something similar because during the test, the unittest does not know where to take the modules to import?

Reply:
Sounds plausible. When I try to debug this kind of issues, I do following:

```
$ nix repl .
> :sh python3.pkgs.pysqlite3 # attribute path of package you are debugging
# That opens bash shell with environment mostly identical to build one
$ mkdir /tmp/sandbox && cd /tmp/sandbox
$ genericBuilder
# See it fail and now you can look around and interactively poke at things until they start working.
```

Reply:
Mmh... @SuperSandro2000 do you have some ideas?

Reply:
> Try deleting the pysqlite3 directory from CWD which might shadow the import

@SuperSandro2000 tried with no luck. If I delete it, the test is not run.

Reply:
@SuperSandro2000 can we continue to manage this please?